10_msi_prediction/nbconverted/lasso_range.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ### LASSO parameter range experiments: microsatellite instability (MSI) prediction
+# 
+# This script is similar to `02_classify_cancer_type/lasso_range_gene.ipynb`, but for MSI prediction across cancer types. MSI information is only included for SKCM (stomach cancer), COAD/READ (colorectal cancer), and UCEC (uterine endometrical carcinoma).
+
+# In[1]:
+
+
+import os
+import itertools as it
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from scipy.stats import pearsonr
+import seaborn as sns
+
+import pancancer_evaluation.config as cfg
+import pancancer_evaluation.utilities.data_utilities as du
+import pancancer_evaluation.utilities.analysis_utilities as au
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# In[2]:
+
+
+base_results_dir = os.path.join(
+    cfg.repo_root, '10_msi_prediction', 'results', 'msi_lasso_range'
+)
+
+training_dataset = 'all_other_cancers'
+results_dir = os.path.join(base_results_dir, training_dataset)
+
+metric = 'aupr'
+nz_cutoff = 5.0
+
+output_plots = False
+output_plots_dir = None
+
+
+# ### Get coefficient information for each lasso penalty
+
+# In[3]:
+
+
+nz_coefs_df = []
+
+# get coefficient info for training dataset specified above
+for coef_info in au.generate_nonzero_coefficients_lasso_range_msi(results_dir):
+    (cancer_type,
+     seed,
+     lasso_param,
+     coefs_list) = coef_info
+    for fold_no, coefs in enumerate(coefs_list):
+        nz_coefs_df.append(
+            [cancer_type, lasso_param, seed, fold_no, len(coefs)]
+        )
+        
+nz_coefs_df = pd.DataFrame(
+    nz_coefs_df,
+    columns=['cancer_type', 'lasso_param', 'seed', 'fold', 'nz_coefs']
+)
+nz_coefs_df.lasso_param = nz_coefs_df.lasso_param.astype(float)
+nz_coefs_df.head()
+
+
+# In[4]:
+
+
+sns.set({'figure.figsize': (12, 5)})
+
+sns.boxplot(
```

Comment:
Since the boxplots on either end of the lasso parameters are so hard to see, could you switch it so the lines of the boxplot are colored, rather than the fill which you can't see on the narrow boxes? Not sure how exactly you'd do that in Seaborn but I'm sure it's possible

Reply:
Good idea! I do think this is easier to follow.

![image](https://user-images.githubusercontent.com/2345877/216643486-d529e6c5-246a-4c16-8359-ce0b10ca8b32.png)